File: C++/1047.cpp
#include <iostream>

int main(){
    int h0, m0, h1, m1;
    int temp;
    int minutes, hours;
    
    std::cin >> h0 >> m0 >> h1 >> m1;
    
    if(h0 == h1 && m0 == m1){
        hours = 24;
        minutes = 0;
    } else {
        int start = (60 * h0) + m0;
        int end = (60 * h1) + m1;
        int d;
        
        if(end > start){
            d = end - start;
        } else { // 1440 = 24h in minutes
            d = (1440 - start) + end;
        }
        
        hours = d / 60;
        minutes = d % 60;
    }

    
    std::cout << "O JOGO DUROU " << hours << " HORA(S) E " << minutes << " MINUTO(S)" << std::endl;
}

File: C#/1047.cs
using System; 

class URI {
    static void Main(string[] args) { 
        string[] v = Console.ReadLine().Split(' ');
        int HI = int.Parse(v[0]);
        int MI = int.Parse(v[1]);
        int HF = int.Parse(v[2]);
        int MF = int.Parse(v[3]);
        int H;
        int M;
        
        if(HI == HF && MI == MF){
            H = 24;
            M = 0;
        } else {
            int I = (HI * 60) + MI;
            int E = (HF * 60) + MF;
            int D;
        
            if(E> I){
                D = E - I;
            } else {
                D=(1440 - I) + E;
            }
            
            H = D / 60;
            M = D % 60;
        }
        
        Console.WriteLine($"O JOGO DUROU {H} HORA(S) E {M} MINUTO(S)");
    }
}